aten/src/ATen/native/quantized/cpu/serialization_versions.h

```diff
@@ -0,0 +1,164 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/core/List.h>
+
+#include <ATen/native/quantized/cpu/fbgemm_utils.h>
+#include <ATen/native/quantized/cpu/qnnpack_utils.h>
+
+#include <tuple>
+
+using SerializationType = std::tuple<
+  std::string /*name of the op that is serialized*/,
+  int64_t /*version*/,
+  std::vector<at::Tensor> /*list of all non-optional tensors*/,
+  std::vector<c10::optional<at::Tensor>> /*list of all optional tensors*/,
+  std::vector<double> /*list of all doubles*/,
+  std::vector<int64_t> /*list of all integers*/
+>;
+
+/* Legacy serialization types */
+
+
+/* Convolution prepacked parameters serialization.
+ *
+ * Files that need to be updated if version changes:
+ * - ATen/native/quantized/cpu/fbgemm_utils.cpp
+ * - jit/passes/onnx/unpack_quantized_weights.cpp
+ *
+ * Version 1 (Legacy)
+ * - Fields:
+ *  1. weight
+ *  2. bias
+ *  3. stride x kSpatialDim
+ *  4. padding x kSpatialDim
+ *  5. dilation x kSpatialDim
+ *  6. groups
+ *
+ * Version 2+
+ * - Fields:
+ *  1. name of the op ("conv")
+ *  2. version number (int64_t)
+ *  3. list of all non-optional tensors (vector<Tensor>)
+ *    1. unpacked weight
+ *  4. list of all optional tensors (vector<optional<Tensor>>)
+ *    1. bias
+ *  5. list of all doubles
+ *    EMPTY
+ *  6. list of all integers (vector<int64_t> of size (1 + 4 * kSpatialDim + 1))
+ *    1. kSpatialDim
+ *    2. stride x kSpatialDim
+ *    3. padding x kSpatialDim
+ *    4. dilation x kSpatialDim
+ *    5. groups
+ */
+const std::string kConvName = "conv";
+constexpr int64_t kConvPackedParamsSerializationVersion = 2;
+template <uint32_t kSpatialDim>
+SerializationType serialize_conv(
+    const c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>>& params) {
+  std::vector<at::Tensor> non_optional;
+  std::vector<c10::optional<at::Tensor>> optional;
+  std::vector<double> doubles;
+  std::vector<int64_t> longs;
+  constexpr int64_t kConvLongsSize = 1 + 4 * kSpatialDim + 1;
+  longs.reserve(kConvLongsSize);
+
+  at::Tensor weight;
+  c10::optional<at::Tensor> bias;
+  std::tie(weight, bias) = params->unpack();
+
+  non_optional.emplace_back(std::move(weight));
+  optional.emplace_back(std::move(bias));
+
+  longs.push_back(kSpatialDim);
+
+  auto stride = params->stride().vec();
+  longs.insert(longs.end(), stride.begin(), stride.end());
+
+  auto padding = params->padding().vec();
+  longs.insert(longs.end(), padding.begin(), padding.end());
+
+  auto dilation = params->dilation().vec();
+  longs.insert(longs.end(), dilation.begin(), dilation.end());
+
+  longs.push_back(params->groups());
+
+
+  return std::make_tuple(
+    kConvName,
+    kConvPackedParamsSerializationVersion,
+    non_optional,
+    optional,
+    doubles,
+    longs
+  );
+}
+
+template <uint32_t kSpatialDim>
+c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>> deserialize_conv(
+    SerializationType state) {
```

Comment:
might be a n00b question.  This looks like the logic to deserialize the v2 format.  Is there logic somewhere to deserialize the v1 format?